src/Microsoft.SqlTools.ServiceLayer/TableDesigner/TableDesignerService.cs

```diff
@@ -57,103 +57,89 @@ public void InitializeService(ServiceHost serviceHost)
             this.ServiceHost.SetRequestHandler(SaveTableChangesRequest.Type, HandleSaveTableChangesRequest);
             this.ServiceHost.SetRequestHandler(DisposeTableDesignerRequest.Type, HandleDisposeTableDesignerRequest);
         }
+        private Task HandleRequest<T>(RequestContext<T> requestContext, Func<Task> action)
+        {
+            Task.Run(async () =>
+            {
+                try
+                {
+                    await action();
+                }
+                catch (Exception e)
+                {
+                    await requestContext.SendError(e);
+                }
+            });
+            return Task.CompletedTask;
```

Comment:
I assume you're purposely not blocking on the task completing here? I'd make it clear if that's the case - and add why you're doing that. 

And to make sure I'm getting this correctly - this will make it so it immediately continues to process other messages but the request context itself is what sends the results/errors and so on the ADS side it'll still be waiting until we send one of those, correct?

Any concern about STS processing multiple designer events simultaneously and causing some race conditions?